framework/src/base/AppFactory.C

```diff
@@ -87,5 +87,13 @@ AppFactory::createShared(const std::string & app_type,
   command_line->addCommandLineOptionsFromParams(parameters);
   command_line->populateInputParams(parameters);
 
+  _app_creation_count[app_type]++;
+
   return _name_to_build_info[app_type]->build(parameters);
 }
+
+size_t
+AppFactory::createdAppCount(const std::string & app_type) const
+{
+  return _app_creation_count.at(app_type);
```

Comment:
should probably return 0 for apps that don't exist/aren't created or return a reasonable error

Reply:
At will throw if the app does not exist. That's technically a valid failure mode, but we usually use find, check the iterator, and issue a more user friendly error message....

Reply:
I'll take `at()` over `operator[]` any day

Reply:
> At will throw if the app does not exist. That's technically a valid failure mode, but we usually use find, check the iterator, and issue a more user friendly error message....

Shouldn't we use `std::map::count` instead of `std::map::find`? Not that this code is performance sensitive but then we save a call to the iterator constructor.

Reply:
According to the comments in https://stackoverflow.com/a/38331636/4493669 you do not save any iteration constructions